PeerSpecSerializer.serialize can fail with NPE

```
21:54:55.885 INFO  [tor.default-dispatcher-12] s.c.n.NetworkController - Unconfirmed connection: (/109.167.229.52:39520, /10.10.2.4:9030) => ConnectionId(remote=/109.167.229.52:39520, local=/10.10.2.4:9030, direction=Incoming)
21:54:55.885 INFO  [tor.default-dispatcher-12] s.c.n.peer.PeerManager - Connection confirmation request: ConnectionId(remote=/109.167.229.52:39520, local=/10.10.2.4:9030, direction=Incoming)
21:54:55.885 INFO  [tor.default-dispatcher-12] s.c.n.NetworkController - Connection confirmed to ConnectionId(remote=/109.167.229.52:39520, local=/10.10.2.4:9030, direction=Incoming)
21:54:55.885 INFO  [tor.default-dispatcher-12] s.c.n.NetworkController - New incoming connection from /109.167.229.52:39520 established (bound to local /10.10.2.4:9030)
21:54:55.886 WARN  [tor.default-dispatcher-12] s.c.n.NetworkController - Stopping child actor failed with: {}
akka.actor.ActorInitializationException: akka://ergoref/user/networkController/$cB: exception during creation
	at akka.actor.ActorInitializationException$.apply(Actor.scala:196)
	at akka.actor.ActorCell.create(ActorCell.scala:661)
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:513)
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:535)
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:295)
	at akka.dispatch.Mailbox.run(Mailbox.scala:230)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException: null
	at scorex.core.network.PeerSpecSerializer.$anonfun$serialize$1(PeerSpec.scala:48)
	at scorex.core.network.PeerSpecSerializer.$anonfun$serialize$1$adapted(PeerSpec.scala:47)
	at scorex.util.serialization.VLQWriter.putOption(VLQWriter.scala:143)
	at scorex.util.serialization.VLQWriter.putOption$(VLQWriter.scala:139)
	at scorex.util.serialization.VLQByteBufferWriter.putOption(VLQByteBufferWriter.scala:9)
	at scorex.util.serialization.VLQByteBufferWriter.putOption(VLQByteBufferWriter.scala:9)
	at scorex.core.network.PeerSpecSerializer.serialize(PeerSpec.scala:47)
	at scorex.core.network.message.HandshakeSpec.serialize(BasicMessagesRepo.scala:267)
	at scorex.core.network.message.HandshakeSpec.serialize(BasicMessagesRepo.scala:252)
	at scorex.core.serialization.ScorexSerializer.toBytes(ScorexSerializer.scala:30)
	at scorex.core.serialization.ScorexSerializer.toBytes$(ScorexSerializer.scala:28)
	at scorex.core.network.message.HandshakeSpec.toBytes(BasicMessagesRepo.scala:252)
	at scorex.core.network.PeerConnectionHandler.handshaking(PeerConnectionHandler.scala:71)
	at scorex.core.network.PeerConnectionHandler.preStart(PeerConnectionHandler.scala:61)
	at akka.actor.Actor.aroundPreStart(Actor.scala:548)
```

--- a/src/main/scala/scorex/core/network/PeerSpec.scala
+++ b/src/main/scala/scorex/core/network/PeerSpec.scala
@@ -43,9 +43,16 @@ class PeerSpecSerializer(featureSerializers: PeerFeature.Serializers) extends Sc
     ApplicationVersionSerializer.serialize(obj.protocolVersion, w)
     w.putShortString(obj.nodeName)
 
-
-    w.putOption(obj.declaredAddress) { (writer, isa) =>
-      val addr = if(isa.getAddress == null) Array.fill[Byte](4){0} else isa.getAddress.getAddress
+    val address = obj.declaredAddress match {
+      case Some(isa) =>
+        if(isa.getAddress == null)
+          None
+        else
+          Some(isa)
+      case None => None
+    }
+    w.putOption(address) { (writer, isa) =>
+      val addr = isa.getAddress.getAddress
       writer.put((addr.size + 4).toByteExact)
       writer.putBytes(addr)
       writer.putUInt(isa.getPort)

--- a/src/main/scala/scorex/core/network/PeerSpec.scala
+++ b/src/main/scala/scorex/core/network/PeerSpec.scala
@@ -45,7 +45,7 @@ class PeerSpecSerializer(featureSerializers: PeerFeature.Serializers) extends Sc
 
 
     w.putOption(obj.declaredAddress) { (writer, isa) =>
-      val addr = isa.getAddress.getAddress
+      val addr: Array[Byte] = Some(isa.getAddress.getAddress).getOrElse(Array.fill[Byte](4){0})
       writer.put((addr.size + 4).toByteExact)
       writer.putBytes(addr)
       writer.putUInt(isa.getPort)

--- a/src/main/scala/scorex/core/network/PeerSpec.scala
+++ b/src/main/scala/scorex/core/network/PeerSpec.scala
@@ -45,7 +45,7 @@ class PeerSpecSerializer(featureSerializers: PeerFeature.Serializers) extends Sc
 
 
     w.putOption(obj.declaredAddress) { (writer, isa) =>
-      val addr: Array[Byte] = Some(isa.getAddress.getAddress).getOrElse(Array.fill[Byte](4){0})
+      val addr = if(isa.getAddress == null) Array.fill[Byte](4){0} else isa.getAddress.getAddress
       writer.put((addr.size + 4).toByteExact)
       writer.putBytes(addr)
       writer.putUInt(isa.getPort)

